### PR TITLE
8332494: java/util/zip/EntryCount64k.java failing with java.lang.RuntimeException: '\\A\\Z' missing from stderr

### PR DESCRIPTION
**Pull request by xmas92**
Improve `java/util/zip/EntryCount64k.java` stderr parsing by ignoring deprecated warnings. Testing non-generational ZGC requires the use of a deprecated option.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Warning
&nbsp;⚠️ Found leading lowercase letter in issue title for `8332494: java/util/zip/EntryCount64k.java failing with java.lang.RuntimeException: '\\A\\Z' missing from stderr`

### Issue
 * [JDK-8332494](https://bugs.openjdk.org/browse/JDK-8332494): java/util/zip/EntryCount64k.java failing with java.lang.RuntimeException: '\\A\\Z' missing from stderr (**Bug** - P4)

### Reviewers
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - **Reviewer**) ⚠️ Review applies to [d25cde7c](https://git.openjdk.org/jdk/pull/19297/files/d25cde7c0f9a5c2651c2847ceaa5bfda35b605e3)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19297/head:pull/19297` \
`$ git checkout pull/19297`

Update a local copy of the PR: \
`$ git checkout pull/19297` \
`$ git pull https://git.openjdk.org/jdk.git pull/19297/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19297`

View PR using the GUI difftool: \
`$ git pr show -t 19297`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19297.diff">https://git.openjdk.org/jdk/pull/19297.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19297#issuecomment-2119782901)

### CODE REVIEW

**Review by jaikiran — APPROVED** [review]
Hello Axel, the change looks OK to me. I was going to suggest `stderrShouldMatch(pattern, boolean ignoreDeprecatedWarning)` as the new method instead of `stderrShouldMatchIgnoreDeprecatedWarnings`, but then I see that there are also methods with names like this in `OutputAnalyzer`. So this change is consistent with those.

`OutputAnalyzer.java` will need a copyright year update before integrating.

**Review by stefank — APPROVED** [review]

**Review by jaikiran — APPROVED** [review]

File: test/lib/jdk/test/lib/process/OutputAnalyzer.java [hunk 1]
```diff
@@ -681,6 +681,24 @@ public OutputAnalyzer stderrShouldMatchIgnoreVMWarnings(String pattern) {
...
     }
 
+    /**
+     * Verify that the stderr contents of output buffer matches the pattern,
+     * after filtering out the Hotspot deprecation warning messages
+     *
+     * @param pattern
+     * @throws RuntimeException If the pattern was not found
+     */
+    public OutputAnalyzer stderrShouldMatchIgnoreDeprecatedWarnings(String pattern) {
```
[the remark this answers is not in the document]
Comment (xmas92) [reply test/lib/jdk/test/lib/process/OutputAnalyzer.java:691]: The name was chosen based on: https://github.com/openjdk/jdk/blob/77c8516085225a04bd5a954197fc5ef7e5c5ee61/test/lib/jdk/test/lib/process/OutputAnalyzer.java#L184

Should I still change it?

[the remark this answers is not in the document]
Comment (dholmes-ora) [reply test/lib/jdk/test/lib/process/OutputAnalyzer.java:691]: No. Sorry I missed the fact the inconsistency had already been introduced. And given that function exists we may as well add the new one too.

**Review by dholmes-ora — APPROVED** [review]
looks good. Thanks

**Comment by xmas92** [comment]
Thanks for the reviews.
/integrate

**Pull request closed without merging**